src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -178,10 +199,116 @@ public static Colony createColony(@NotNull final World w, final BlockPos pos, @N
         colony.getStatsManager().triggerAchievement(ModAchievements.achievementTownhall);
         Log.getLogger().info(String.format("New Colony Id: %d by %s", colony.getID(), player.getName()));
 
+        ColonyManager.notifyChunksInRange(colony.getWorld(), true, colony.getID(), colony.getCenter(), colony.getDimension());
         ColonyManager.markDirty();
         return colony;
     }
 
+    /**
+     * Notify all chunks in the range of the colony about the colony.
+     * @param world the world of the colony.
+     * @param add remove or add
+     */
+    public static void notifyChunksInRange(final World world, final boolean add, final int id, final BlockPos center, final int dimension)
+    {
+        final Chunk centralChunk = world.getChunkFromBlockCoords(center);
+        if(centralChunk.getCapability(CLOSE_COLONY_CAP, null).getOwningColony() == id && add)
+        {
+            return;
+        }
+        final IColonyTagCapability cap = centralChunk.getCapability(CLOSE_COLONY_CAP, null);
+
+        if(add)
+        {
+            cap.setOwningColony(id);
+            cap.addColony(id);
+        }
+        else
+        {
+            cap.removeColony(id);
+        }
+        centralChunk.markDirty();
+        MineColonies.getNetwork().sendToAll(new UpdateChunkCapabilityMessage(centralChunk.getCapability(CLOSE_COLONY_CAP, null), centralChunk.x, centralChunk.z));
+
+        final int chunkX = centralChunk.x;
+        final int chunkZ = centralChunk.z;
+
+        final int range = Configurations.gameplay.workingRangeTownHallChunks;
+        final int buffer = Configurations.gameplay.townHallPaddingChunk;
+
+        final int maxRange = range * 2 + buffer;
+        @NotNull final File chunkDir = new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), CHUNK_INFO_PATH);
+        Utils.checkDirectory(chunkDir);
+
+        for(int i = chunkX - maxRange; i <= chunkX + maxRange; i++)
+        {
+            for(int j = chunkZ - maxRange; j <= chunkZ + maxRange; j++)
+            {
+                if(i >= chunkX - range && j >= chunkZ - range && i <= chunkX + range && j <= chunkZ + range)
+                {
+                    @NotNull final File file = new File(chunkDir, String.format(FILENAME_CHUNK, i, j, dimension));
+                    @NotNull final ChunkLoadStorage newStorage = new ChunkLoadStorage(id, ChunkPos.asLong(i,j), add, dimension);
+                    if (file.exists())
+                    {
+                        @Nullable final NBTTagCompound chunkData = loadNBTFromPath(file);
+                        final ChunkLoadStorage storage = new ChunkLoadStorage(chunkData);
+                        storage.merge(newStorage);
+
+                        if(storage.isEmpty())
+                        {
+                            file.delete();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Do something with the "boolean" value returned by "delete". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS899)